scalyr_agent/builtin_monitors/kubernetes_monitor.py

```diff
@@ -159,9 +161,27 @@
                      convert_to=int, default=20)
 
 define_config_option( __monitor__, 'k8s_parse_json',
-                      'Optional (defaults to True). If True, the log files will be parsed as json before uploading to the server '
-                      'to extract log and timestamp fields.  If False, the raw json will be uploaded to Scalyr.',
-                      convert_to=bool, default=True, env_aware=True)
+                      'Deprecated, please use `k8s_parse_format`. If set, and True, then this flag will override the `k8s_parse_format` to `auto`. '
+                      'If set and False, then this flag will override the `k8s_parse_format` to `raw`.',
+                      convert_to=bool, default=None)
+
+define_config_option( __monitor__, 'k8s_parse_format',
+                      'Optional (defaults to `auto`). Valid values are: `auto`, `json`, `cri` and `raw`. '
+                      'If `auto`, the monitor will try to detect the format of the raw log files, '
+                      'e.g. `json` or `cri`.  Log files will be parsed in this format before uploading to the server '
+                      'to extract log and timestamp fields.  If `raw`, the raw contents of the log will be uploaded to Scalyr. '
+                      '(Note: An incorrect setting can cause parsing to fail which will result in raw logs being uploaded to Scalyr, so please leave '
+                      'this as `auto` if in doubt.)',
+                      convert_to=str, default='auto', env_aware=True)
+
+define_config_option( __monitor__, 'k8s_always_use_cri',
+                     'Optional (defaults to False). If True, the kubernetes monitor will always try to read logs using the container runtime interface '
+                     'even when the runtime is detected as docker',
+                     convert_to=bool, default=False, env_aware=True)
+
+define_config_option( __monitor__, 'k8s_cri_query_filesystem',
+                     'Optional (defaults to False). If True, then when in CRI mode, the monitor will query the filesystem for the list of active containers, rather than the Kubelet API.  This is useful in environments where the Kubelet API has been disabled.',
```

Comment:
Recommend changing the English a bit for correctness (the existing text is from a previous iteration that did not auto-switch to filesystem I think):

"If True, then when in CRI mode, the monitor will only query the filesystem for the list of active containers, rather than **first querying** the Kubelet API.  This is a useful optimization when the Kubelet API is known to be disabled."